src/Data/Geometry/Structure/Feature.hs

```diff
@@ -67,10 +73,22 @@ instance LikeJSArray "Object" FeatureCollection where
     {-# INLINE fromJSArray #-}
     fromJSArray = js_JSArrayToFC
 
+-- | JSON GeometryInput
+newtype GeometryInput = GeometryInput JSVal
+instance LikeJS "Object" GeometryInput where
+  asJSVal = js_deleteGiTimestamp
+
 ----------------------------------------------------------------------------------------------------
 -- Some Functions
 ----------------------------------------------------------------------------------------------------
 
+data ParsedGeometryInput x = ParsedGeometryInput
+  { pgiFeatureCollection :: FeatureCollection
+  , pgiErrors            :: JS.Array JSString
+  , pgiLatLonAlt         :: Maybe (Vector 3 x)
```

Comment:
Better create a separate data type:
data GeoLocation = GeoLocCoords { lat :: Float, lon :: Float, alt :: Float } | GeoLocSRID Int
and then you can have pgiLoc :: Maybe GeoLocation
This way we avoid ambiguous situation when both coords and srid are present